### PR TITLE
[zlib, minizip] Update to 1.2.13, and embed the ZLIB_DLL setting.

### PR DESCRIPTION
**Pull request by BillyONeal**
This picks up the official fix for CVE-2022-37434.

* Fixed the patches to zlib apply cleanly to this version
* Simplified the removal of examples to unconditional (after all, we're patching anyway)
* Reduced some of the static-or-shared patch to avoid repeating the line of code put into the if block
* Combined the MinGW and Android patches since they were touching the same bits
* Fixes https://github.com/microsoft/vcpkg/pull/26885 . Which matches ZLIB_DLL to the setting selected by the triplet. (Thanks @dg0yt !)

### CODE REVIEW

**Comment by dg0yt** [comment]
Please integrate #26885.

File: ports/zlib/0003-build-static-or-shared-not-both.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                             -i ${CMAKE_CURRENT_SOURCE_DIR}/win32/zlib1.rc)
++    if(BUILD_SHARED_LIBS)
+     set(ZLIB_DLL_SRCS ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj)
+ endif(MINGW)
++    endif()
+ 
+-add_library(zlib SHARED ${ZLIB_SRCS} ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
+-add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
++add_library(zlib ${ZLIB_SRCS} ${ZLIB_ASMS} ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
+ set_target_properties(zlib PROPERTIES DEFINE_SYMBOL ZLIB_DLL)
```
Comment (ras0219-msft) [inline ports/zlib/0003-build-static-or-shared-not-both.patch:29]: Does this line need to be guarded with `if(BUILD_SHARED_LIBS)` as well?

**Review by ras0219-msft — DISMISSED** [review]
I agree that we should include https://github.com/microsoft/vcpkg/pull/26885 if possible.

File: ports/zlib/0003-build-static-or-shared-not-both.patch (hunk 1 above)
Comment (BillyONeal) [reply ports/zlib/0003-build-static-or-shared-not-both.patch:29]: I admit I didn't pay that close attention to what the patch was doing, this is just cmake-dont-build-more-than-needed.patch retargeted for this base commit. Will consider with the other PR...

**Comment by CromFr** [comment]
I think the download URL for zlib should also be updated to `https://zlib.net/fossils/zlib-<version>.tar.gz`.

All my VCPKG builds are currently failing because `https://www.zlib.net/zlib-1.2.12.tar.gz` returns 404

**Comment by BillyONeal** [comment]
I cancelled the PR build because I'm ensuring I understand

> Please integrate #26885.

I do love that change.

**Comment by BillyONeal** [comment]
@Neumann-A It looks like the minizip patch added in https://github.com/microsoft/vcpkg/pull/25960/ is effectively already present in 1.2.13 so I have removed it.

File: ports/zlib/0003-build-static-or-shared-not-both.patch (hunk 1 above)
Comment (BillyONeal) [reply ports/zlib/0003-build-static-or-shared-not-both.patch:30]: Agreed, also guarded SOVERSION.

**Review by ras0219-msft — APPROVED** [review]

**Pull request merged**